os.cpp

```diff
@@ -1,380 +0,0 @@
-				}
-			} else {
-				cout << "SYS > ";
-			}
-
-		// If necessary, take input from command line
-		if ((!running  && !instrLoaded) || currentUser.id == sys) {
-			string input;
-			getline(cin, input);
-			switch (cmdToInt(input)) {
-				case 0: // "run"
-					if (currentUser.id != sys) {
-						machine.IR = main_memory[machine.PC];
-						// If memory is free, run as usual.
-						if (semwait() == false) {
-							cout << endl << titleFiller;
-							cout << endl << "USER " << currentUser.id << " REQUESTED MEM " \
-								<< "ACCESS AND ACCESS WAS GRANTED." << endl << "USER MAY " \
-								<< "EXECUTE INSTRUCTIONS.";
-							cout << endl << titleFiller << endl << endl;
-							currentUser.running = true;
-							interpreter();
-							// If last instruction executed was HLT
-							if (machine.IR == 61440) {
-								// Signal semaphore is unlocked
-								semsignal();
-								currentUser.running = false;
-								// Exit loop (current user is done)
-								switchTime = sysclock;
-							}
-						// If memory is locked, dispatch to blockedQueue, load next user
-						// Reset switchtime (i.e. reset the loop) so next user gets
-						// 3 ticks.
-						} else {
-							switchTime = sysclock + 3;
-						}
-					} else {
-						cout << "Invalid command for system" << endl;
-					}
-					break;
-				case 1: // "dmp"
-					if (currentUser.id == sys) {
-						dump();
-					} else {
-						cout << "Invalid command for users" << endl;
-					}
-					break;
-				case 2: // "nop"
-					// Exit loop (current user is done)
-					switchTime = sysclock;
-					break;
-				case 3: // "stp"
-					if (currentUser.id == sys) {
-						exit(EXIT_SUCCESS);
-					} else {
-						cout << "Invalid command for users" << endl;
-					}
-					break;
-				default:
-					cout << "Invalid command: " << input << endl;
-					break;
-				}
-			}
-		}
-	}
-}
-
-// Read program into memory
-void readFile()
-{
-	// This method will need to be rewritten entirely using c++ streams
-	ifstream infile("part2.dat");
-	int i = 0;
-	short unsigned int current;
-	string line;
-	while (std::getline(infile, line))
-	{
-		if (line == "*") {
-			i = 100;
-			continue;
-		}
-		char * ptr;
-		current = strtol(line.c_str(), & ptr, 2);
-		main_memory[i] = current;
-		i++;
-	}
-}
-
-// Convert a command as a string into an integer value to simplify
-// switch statements using commands.
-int cmdToInt(string cmd)
-{
-	if (my_strcasecmp(cmd, "run")) return 0;
-	else if (my_strcasecmp(cmd, "dmp")) return 1;
-	else if (my_strcasecmp(cmd, "nop")) return 2;
-	else if (my_strcasecmp(cmd, "stp")) return 3;
-	else return -1;
-}
-
-// Case-insensitive string comparison
-bool my_strcasecmp(string str1, string str2)
-{
-	int length1 = str1.length();
-	for (int i = 0; i < length1; i++) {
-		char &c = str1.at(i);
-		c = tolower(c);
-	}
-	int length2 = str2.length();
-	for (int i = 0; i < length2; i++) {
-		char &c = str2.at(i);
-		c = tolower(c);
-	}
-	return ((str1 == str2) ? true : false);
-}
```

Comment:
This function should really go in a separate file called utils.

We could also keep any bin2hex or hex2bin functions in this utils file.